src/software/thunderscope/replay/proto_player.py

```diff
@@ -85,22 +107,45 @@ def __sort_replay_chunks(file_path: str):
             replay_index, _ = tail.split(".")
             return int(replay_index)
 
-        self.sorted_chunks = sorted(replay_files, key=__sort_replay_chunks)
+        return sorted(replay_files, key=__sort_replay_chunks)
 
-        # We can get the total runtime of the log from the last entry in the last chunk
-        last_chunk_data = ProtoPlayer.load_replay_chunk(self.sorted_chunks[-1])
-        try:
-            self.end_time, _, _ = ProtoPlayer.unpack_log_entry(last_chunk_data[-1])
-        except DecodeError:
-            self.end_time, _, _ = ProtoPlayer.unpack_log_entry(last_chunk_data[-2])
-        logging.info(
-            "Loaded log file with total runtime of {:.2f} seconds".format(self.end_time)
-        )
+    @staticmethod
+    def is_log_entry_corrupt(log_entry) -> bool:
+        """
+        Check to see if we have can unpack the log entry
+
+        :param log_entry: the log entry we are checking
+        :return: False if we could unpack the log entry, True otherwise
+        """
+        try: 
+            _ = ProtoPlayer.unpack_log_entry(log_entry)
+            return False
+        except Exception: 
+            logging.warning("There exist log entries that are corrupt.")
```

Comment:
This error seems to get spammed with normal replay logs but the replay plays fine. Any idea what might be happening?

Reply:
I have no idea. It seems to work in my branch. Would you mind attach the proto replay?

Reply:
Also, this is by design. I want to warn the user that some of the entries are corrupted. Thus, proto player would play fine with the corrupted log entries, which are ignored when being played.

Reply:
That makes sense to be intended, though it is weird that we get them logged even for normal replays taken from AI vs AI. Here's the replay I got. Could you see why it's logging those messages? Maybe we're not importing a specific protobuf type so it can't parse it in the replay.
[proto_2024_06_13_203104.zip](https://github.com/user-attachments/files/15827156/proto_2024_06_13_203104.zip)